### ISSUE TITLE
12-nlp-url-spam.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/benjaminmullermiranda-a11y-Machine_Learning#7]
@benjaminmullermiranda-a11y  excelente proyecto bien estructurado. 

**Problemas:**

**1. GridSearchCV usa scoring='accuracy' (debería ser 'f1')**

**Por qué es problema:** En spam detection, el desbalance es crítico (ej: 5% spam). Un modelo que predice "todo no-spam" tiene accuracy alta pero es inútil. **F1 score** balancea precisión/recall. `accuracy` favorece la clase mayoritaria.

**Cómo arreglarlo:**
```python
grid_search = GridSearchCV(
    SVC(random_state=42),
    param_grid,
    cv=5,
    scoring='f1',  # Cambiar de 'accuracy' a 'f1'
    n_jobs=-1,
    verbose=1
)
```

---

Sugerencia:

**2. Sin matriz de confusión**

**Por qué es problema:** `classification_report` muestra números (precision, recall, f1), pero una matriz 2x2 **visualiza qué tipo de error comete**: FP (spam legítimo marcado) vs FN (spam no detectado).

**Cómo arreglarlo:**
```python
from sklearn.metrics import confusion_matrix
import seaborn as sns

cm = confusion_matrix(y_test, y_pred_best)

fig, ax = plt.subplots(figsize=(8, 6))
sns.heatmap(cm, annot=True, fmt='d', cmap='Blues', 
            xticklabels=['No Spam', 'Spam'],
            yticklabels=['No Spam', 'Spam'],
            ax=ax, cbar_kws={'label': 'Count'})
ax.set_title('Matriz de Confusión - SVM Optimizado')
ax.set_ylabel('Real')
ax.set_xlabel('Predicción')
plt.tight_layout()
plt.show()
```
